### PR TITLE
Add segmentation module

### PR DESCRIPTION
**Pull request by roshankern**
@gwaybio This should be good for review! I will raise an issue for the `no corresponding object` error.

### CODE REVIEW

File: 2.segment_nuclei/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# 2. Segment Nuclei
+
+In this module, we present our pipline for segmenting nuclei from the mitosis movies.
```
Comment (gwaybio) [inline 2.segment_nuclei/README.md]:
```suggestion
In this module, we present our pipeline for segmenting nuclei from the mitosis movies.

```

File: 2.segment_nuclei/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# 2. Segment Nuclei
+
+In this module, we present our pipline for segmenting nuclei from the mitosis movies.
+
+### Segmentation
+
+We use CellPose algorithm to segment the nuclei from each mitosis movie. 
+CellPose was first introduced in [Stringer, C., Wang, T., Michaelos, M. et al., 2020](https://doi.org/10.1038/s41592-020-01018-x) and we use the [python implementation](https://github.com/mouseland/cellpose).
```
Comment (gwaybio) [inline 2.segment_nuclei/README.md]:
```suggestion
We use the CellPose nucleus model to segment the nuclei from each mitosis movie. 
CellPose was first introduced in [Stringer, C., Wang, T., Michaelos, M. et al., 2020](https://doi.org/10.1038/s41592-020-01018-x) and we use the [python implementation](https://github.com/mouseland/cellpose).

Stringer et al. trained the CellPose nucleus model on a diverse set of nuclei images and is therefore a good selection for our use case.
The CellPose python implementation was particularly useful for building reproducible pipelines.
```

Is what I added accurate? It is good to justify our selection (there are _many_ different options here).

**Review by gwaybio — CHANGES_REQUESTED** [review]
What are all of the `.dat.gz` files? They are very small, but there are a ton!

Also, I don't see the `segment_training_data.py` file. Is this for a future PR?

File: 2.segment_nuclei/README.md (hunk 2 above)
Comment (roshankern) [reply 2.segment_nuclei/README.md]: Yes, that is accurate. Thank you!

**Comment by roshankern** [comment]
The `.dat.gz` files contain the bounding box coordinates for the labeled Mitocheck data which we use to label the CellPose-derived nuclei locations.

Not sure why you cant see the `segment_training_data.py` file. This was added in https://github.com/WayScience/mitocheck_data/pull/5/commits/a054aad9723341c7706fdcf482c70d8fae505f49.
Is it possible you are viewing all changes and the `diff` only shows the first 3000 file changes?
If so the `.dat.gz` files likely take up those spots so you don't see the file added after the `.dat.gz` files.

**Comment by gwaybio** [comment]
Got it! Thanks for clarifying. Only three other comments:

1. It looks like you're using the `cyto` cellpose model. I thought that we were using the `nucleus` model? If we're using the cyto, we need to update the README and justify why not nuclei
2. I'm seeing a keyboard interrupt in the jupyter notebook - would it take too long to re-run the full notebook to completion to prevent this?
3. Do you have documentation about your troubleshooting solutions for installing cellpose? Did you have to create a new conda environment? I imagine that if not yet, we will have to make a separate one soon for the DeepProfiler implementation

**Comment by roshankern** [comment]
Comments 1 & 2 should be resolved in the latest commit, let me know what you think.

Regarding comment 3, the issues were solved by installing CellPose from `conda-forge` instead of `pip`. I did not create a separate environment for this module but think that might be a good idea. I have opened an issue for this (https://github.com/WayScience/mitocheck_data/issues/6) which we can discuss during our in-person meeting.

**Review by gwaybio — APPROVED** [review]

**Pull request merged**